modules/core/02-client/migrations/v7/store.go

```diff
@@ -0,0 +1,215 @@
+package v7
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	host "github.com/cosmos/ibc-go/v6/modules/core/24-host"
+	"github.com/cosmos/ibc-go/v6/modules/core/exported"
+	solomachine "github.com/cosmos/ibc-go/v6/modules/light-clients/06-solomachine"
+	ibctm "github.com/cosmos/ibc-go/v6/modules/light-clients/07-tendermint"
+)
+
+// Localhost is the client type for a localhost client. It is also used as the clientID
+// for the localhost client.
+const Localhost string = "09-localhost"
+
+// MigrateStore performs in-place store migrations from ibc-go v6 to ibc-go v7.
+// The migration includes:
+//
+// - Migrating solo machine client states from v2 to v3 protobuf definition
+// - Pruning all solo machine consensus states
+// - Removing the localhost client
+// - Asserting existing tendermint clients are properly registered on the chain codec
+func MigrateStore(ctx sdk.Context, storeKey storetypes.StoreKey, cdc codec.BinaryCodec) error {
+	store := ctx.KVStore(storeKey)
+
+	if err := handleSolomachineMigration(ctx, store, cdc); err != nil {
+		return err
+	}
+
+	if err := handleTendermintMigration(ctx, store, cdc); err != nil {
+		return err
+	}
+
+	if err := handleLocalhostMigration(ctx, store, cdc); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// handleSolomachineMigration iterates over the solo machine clients and migrates client state from
+// protobuf definition v2 to v3. All consensus states stored outside of the client state are pruned.
+func handleSolomachineMigration(ctx sdk.Context, store sdk.KVStore, cdc codec.BinaryCodec) error {
+	clients, err := collectClients(ctx, store, exported.Solomachine)
+	if err != nil {
+		return err
+	}
+
+	for _, clientID := range clients {
+		clientPrefix := []byte(fmt.Sprintf("%s/%s/", host.KeyClientStorePrefix, clientID))
+		clientStore := prefix.NewStore(store, clientPrefix)
+
+		bz := clientStore.Get(host.ClientStateKey())
+		if bz == nil {
+			return clienttypes.ErrClientNotFound
+		}
+
+		any := &codectypes.Any{}
+		if err := cdc.Unmarshal(bz, any); err != nil {
+			return sdkerrors.Wrap(err, "failed to unmarshal client state bytes into solo machine client state")
+		}
+
+		clientState := &ClientState{}
+		if err := cdc.Unmarshal(any.Value, clientState); err != nil {
+			return sdkerrors.Wrap(err, "failed to unmarshal client state bytes into solo machine client state")
+		}
+
```

Comment:
I think we normally favour unmarshalling into types like this in other parts of the codebase.
But I see the `ClientState` is a passed as a pointer type to `migrateSolomachine` also.

Happy to leave as is seeing as this is migration code anyways.

```suggestion
		var any codectypes.Any
		if err := cdc.Unmarshal(bz, &any); err != nil {
			return sdkerrors.Wrap(err, "failed to unmarshal client state bytes into solo machine client state")
		}

		var clientState ClientState
		if err := cdc.Unmarshal(any.Value, &clientState); err != nil {
			return sdkerrors.Wrap(err, "failed to unmarshal client state bytes into solo machine client state")
		}

```

Reply:
applied!